java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/CuVSResourcesImpl.java

```diff
@@ -46,19 +45,22 @@ public class CuVSResourcesImpl implements CuVSResources {
    */
   public CuVSResourcesImpl(Path tempDirectory) throws Throwable {
     this.tempDirectory = tempDirectory;
-    arena = Arena.ofShared();
-    resourcesMemorySegment = arena.allocate(cuvsResources_t);
-    int returnValue = cuvsResourcesCreate(resourcesMemorySegment);
-    checkCuVSError(returnValue, "cuvsResourcesCreate");
+    this.arena = Arena.ofShared();
+    try (var localArena = Arena.ofConfined()) {
+      var resourcesMemorySegment = localArena.allocate(cuvsResources_t);
+      int returnValue = cuvsResourcesCreate(resourcesMemorySegment);
+      checkCuVSError(returnValue, "cuvsResourcesCreate");
+      resourceHandle = resourcesMemorySegment.get(cuvsResources_t, 0);
```

Comment:
Ok, I think I see where you're going with this.  It makes more sense to keep the resource handle handy than the `MemorySegment` that contains it.